### PR TITLE
[CI-APP][CIAPP-240] Trace Command

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What and why?

[Jira ticket](https://datadoghq.atlassian.net/browse/CIAPP-240)

Create spans associated to CI commands.

An example trace can be found [here](https://app.datadoghq.com/apm/trace/1445452801424440346?spanID=5198930442363958063&env=testing-juan&sort=time&colorBy=service&graphType=flamegraph&shouldShowLegend=true). The corresponding `JenkinsFile` can be found in https://github.com/DataDog/datadog-ci/blob/juan-fernandez/ci-app-trace-command-test-jenkins/Jenkinsfile

For the moment we only support Jenkins for continuing the trace, though this command could be used anywhere (the spans would stand alone). 

### How?

* `spawn` command and create span with exit code and some other metadata. 
* Use `dd-trace` to create the span with the extracted context (only Jenkins supported at the moment). 

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit)
- [ ] Feature or bugfix MUST have appropriate tests (integration)

### CODE REVIEW

File: package.json
Comment (sdeprez) [inline package.json:41]: Let's prefer fixed version (without `^`)

File: src/helpers/interfaces.ts
Comment (sdeprez) [inline src/helpers/interfaces.ts:12]: Inconsistent case, let's rename this one `commitSha`. It's a breaking change but it's ok since it was never used anywhere in Synthetics UI *but* it still requires a change of our schemas so it doesn't break the intake -> https://datadoghq.atlassian.net/browse/SSV-252

File: src/helpers/tags.ts
Comment (sdeprez) [inline src/helpers/tags.ts:19]: Not a big deal but I'm not sure those constants have their place in `helpers`, it looks like they are specific to the `trace` command?

**Review by sdeprez — APPROVED** [review]

**Review by m-rousse — APPROVED** [review]
LGTM modulo Sdeprez's comments!

**Review by yoannmoinet — APPROVED** [review]
LGTM

File: LICENSE-3rdparty.csv
Comment (yoannmoinet) [reply LICENSE-3rdparty.csv:31]: But it will fail the 3rdparty test if we change this, right @m-rousse ?
Not sure if we should change this or not.

Comment (juan-fernandez) [reply LICENSE-3rdparty.csv]: it seems to pass without the comma

File: src/helpers/interfaces.ts
Comment (juan-fernandez) [reply src/helpers/interfaces.ts]: ok so to confirm this is blocked until https://datadoghq.atlassian.net/browse/SSV-252 is done, right?

Comment (juan-fernandez) [reply src/helpers/interfaces.ts]: ok yeah tests are failing: https://github.com/DataDog/datadog-ci/runs/1320388975?check_suite_focus=true

Comment (yoannmoinet) [reply src/helpers/interfaces.ts]: Shouldn't this be done in a separate PR @sdeprez ?

Comment (juan-fernandez) [reply src/helpers/interfaces.ts]: that makes sense. If @sdeprez agrees I'll separate this into a different PR

File: LICENSE-3rdparty.csv
Comment (m-rousse) [reply LICENSE-3rdparty.csv]: The script does not check the number of columns (nor that the license is indeed the right one) but only that there is a line for each dependency listed in the package.json. That's why is does not fail in either case.

File: src/helpers/interfaces.ts
Comment (sdeprez) [reply src/helpers/interfaces.ts]: Whatever works yes, it can be done in another PR to unlock this one (as long as both PRs are done before someones uses this new command so it has proper naming `commitSha` since the beginning!)

**Pull request merged**
